python/ray/data/tests/test_stats.py

```diff
@@ -1208,7 +1208,7 @@ def sleep_three(x):
 
 
 def test_stats_actor_iter_metrics():
-    ds = ray.data.range(1e6).map_batches(lambda x: x)
```

Comment:
Should we check in `RangeDatasource` that `n` is an int? Using scientific notation results in a float

Reply:
What happens if we pass a float to `range`? Do we get the `StatsActor` error?

Reply:
`StatsActor` works with float, it's just for the `BlockMetadata.size_bytes` typing

Reply:
I see.

> Should we check in RangeDatasource that n is an int? Using scientific notation results in a float

What do you have in mind? Like adding an `assert isinstance(n, int)` or raise a `ValueError`?

Reply:
Yeah I think raising a `ValueError` could work here. Maybe it's too harsh? We could also just try casting here. What do you think?

Reply:
maybe just convert the input n to int. scientific notation is convenient to use. raising errors seems to cause unnecessary inconvenience

Reply:
Yeah, +1 to casting input to `int`. If there's some loss in data (e.g, going from 10.6 to 10), we could raise a warning.

While defensive programming is good, I think it's also a good practice to define errors out of existence (Chapter 10 of [Philosophy of Software Design](https://milkov.tech/assets/psd.pdf))